src/signature.c

```diff
@@ -82,20 +100,38 @@ bool initialize_signature(void)
 
 	ret = validate_certificate();
 	if (ret) {
-		printf("Failed to verify certificate: %s\n", X509_verify_cert_error_string(ret));
 		if (ret == X509_V_ERR_CERT_NOT_YET_VALID) {
-			/* If we can retrieve an approx. good system time, report out to user */
-			if (stat("/usr/lib/os-release", &statt) != -1) {
-				mod_sec = statt.st_mtim.tv_sec;
-				char timebuf[30];
-				alttime = localtime(&mod_sec);
-				strftime(timebuf, sizeof(timebuf), "%F", alttime);
-				printf("System clock should be at least %s\n", timebuf);
+			time_t currtime;
```

Comment:
Yea this part is the most ugly/difficult.

So the matrix:
if system time is good and cert is good -> everything works
if system time is good and cert is bad -> need to figure out how to change this code do the rightish thing
if system time is bad and cert is good -> this code as is works
if system time is bad and cert is bad -> this code as is works

So while not breaking the bad system time case we need to handle the good system time case with bad cert.

We can do some kind of time delta between system time and file time and see if that is some amount off. Possibly could account for the cert expiration dates but I think figuring out the cert being used is really hard here and maybe makes this more fragile than the simple date delta from file and system time.